oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/hybrid/ManyFacetsTest.java

```diff
@@ -0,0 +1,280 @@
+    private void runAsyncIndex() {
+        AsyncIndexUpdate async = (AsyncIndexUpdate) WhiteboardUtils.getService(wb, Runnable.class, new Predicate<Runnable>() {
+            @Override
+            public boolean test(@Nullable Runnable input) {
+                return input instanceof AsyncIndexUpdate;
+            }
+        });
+        assertNotNull(async);
+        async.run();
+        if (async.isFailing()) {
+            fail("AsyncIndexUpdate failed");
+        }
+        root.refresh();
+    }
+    
+    @Test
+    public void facet() throws Exception {
+        // Explicitly setting following configs to run DelayedLuceneFacetProvider and a thread sleep of 50 ms in refresh readers. Refer: OAK-8898
+        System.setProperty(LucenePropertyIndex.OLD_FACET_PROVIDER_CONFIG_NAME, "false");
+        // The variable is static final so once set it remains same for all tests and which will lead to slow execution
+        // of other tests as this add a sleep of specified milliseconds in refresh reader method in LuceneIndexNodeManager.
+       // System.setProperty(LuceneIndexNodeManager.OLD_FACET_PROVIDER_TEST_FAILURE_SLEEP_INSTRUMENT_NAME, "40");
+        Thread.currentThread().setName("main");
+        String idxName = "hybridtest";
+        Tree idx = createIndex(root.getTree("/"), idxName);
+        TestUtil.enableIndexingMode(idx, FulltextIndexConstants.IndexingMode.NRT);
+        setTraversalEnabled(false);
+        root.commit();
+        jcr = new Jcr(oak);
+        jcrRepo = jcr.createRepository();
+        createSmallDataset(0);
+        clock.waitUntil(clock.getTime() + REFRESH_DELTA + 1);
+        root.commit();
+        runAsyncIndex();
+        createSmallDataset(2);
+        clock.waitUntil(clock.getTime() + REFRESH_DELTA + 1);
+        root.commit();
+        Session anonSession = jcrRepo.login(new GuestCredentials());
+        qm = anonSession.getWorkspace().getQueryManager();
+        String facetList = "";
+        for (int i = 0; i < FACET_COUNT; i++) {
+            if (i > 0) {
+                facetList += ", ";
+            }
+            facetList += "[rep:facet(foo" + i + ")]";
+        }
+        String queryString = "SELECT " + facetList + 
+                " FROM [nt:base] WHERE [cons] = 'val'";
+        Query q = qm.createQuery(queryString, SQL2);
+        QueryResult qr = q.execute();
+        try {
+            RowIterator it = qr.getRows();
+            assertTrue(it.hasNext());
+            while (it.hasNext()) {
+                Row r = it.nextRow();
+                for (int i = 0; i < qr.getColumnNames().length; i++) {
+                    String columnName = qr.getColumnNames()[i];
+                    String v = r.getValue(columnName).getString();
+                    JsonObject json = JsonObject.fromJson(v, true);
+                    for (int j = 0; j < NUM_LABELS; j++) {
+                        String n = json.getProperties().get("foo" + i + "x" + j);
+                        assertTrue(n != null);
```

Comment:
`assertNotNull` is a better option here
```suggestion
                        assertNotNull(n);
```